# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1471557540 -7200
#      Thu Aug 18 23:59:00 2016 +0200
# Node ID 788f554ac87d4e4d2150a45e16249d862aba3d5c
# Parent  d71ae2cd510bc224e8c80c85a603df69d8237a13
clabel.m: Fix computation of label rotation (bug #48814).

* __clabel__.m: use absolute coordinates to compute the label rotation angle.

diff --git a/scripts/plot/appearance/__clabel__.m b/scripts/plot/appearance/__clabel__.m
--- a/scripts/plot/appearance/__clabel__.m
+++ b/scripts/plot/appearance/__clabel__.m
@@ -96,18 +96,18 @@ function h = __clabel__ (c, v, hparent, 
       j = find (cumd > tagpos, 1);
       if (isempty (j))
         j = clen;
       endif
       tpos = sum (c(:,i+j-1:i+j), 2) / 2;
 
       if (   tpos(1) != xmin && tpos(1) != xmax
           && tpos(2) != ymin && tpos(2) != ymax)
-        trot = 180 / pi * atan2 (diff (c(2,i+j-1:i+j)),
-                                 diff (c(1,i+j-1:i+j)));
+        trot = 180 / pi * atan2 (diff (c(2,i+j-1:i+j)) * yspacing,
+                                 diff (c(1,i+j-1:i+j)) * xspacing);
         if (abs (trot) > 90)
           trot += 180;
         endif
         if (ischar (z))
           ht = text (tpos(1), tpos(2), clev, tlabel, "rotation", trot,
                      "horizontalalignment", "center", "userdata", clev,
                      "parent", hparent, varargin{:});
         elseif (! isempty (z))
